### PR TITLE
full project code refactored

### PR DESCRIPTION
**Pull request by BogdanKalysh**

### CODE REVIEW

File: DAL/headers/models/user.h [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
+#include <constants.h>
 
 class User
 {
     int id;
     QString name;
     QString password;
     QString email;
-    int balance;
+//    int balance;
```
Comment (AndriyFedorenko) [inline DAL/headers/models/user.h:12]: remove commented code

File: DAL/sources/models/transaction.cpp [hunk 2]
```diff
@@ -62,55 +62,42 @@ QString Transaction::getType()
...
 
-QString Transaction::setColor(const QColor &color)
+void Transaction::setColor(const QColor &color)
 {
     this->color = color;
-    return "Success";
 }
 
-QString Transaction::setType(const QString type)
+void Transaction::setType(const QString type)
```
Comment (AndriyFedorenko) [inline DAL/sources/models/transaction.cpp:100]: please use reference

File: DAL/sources/repositories/categoryrepository.cpp [hunk 3]
```diff
@@ -30,76 +31,75 @@ bool CategoryRepository::update(const Category &object)
...
-        else
-        {
-            qDebug() << query.lastError();
-            db.close();
-            return false;
-        }
 
-    } else {
-        qDebug() << db.lastError();
+        qDebug() << query.lastError();
```
Comment (AndriyFedorenko) [inline DAL/sources/repositories/categoryrepository.cpp:79]: This should be logged into log file

File: budget_manager_backend/sources/applicationservice.cpp [hunk 4]
```diff
@@ -30,24 +30,24 @@ class Socket: public Poco::Net::Socket
 };
 }
 
+//ApplicationService::ApplicationService()
```
Comment (AndriyFedorenko) [inline budget_manager_backend/sources/applicationservice.cpp:33]: please remove it

File: budget_manager_backend/sources/handlers/categoryhandler.cpp [hunk 5]
```diff
@@ -16,82 +16,93 @@ AbstractHandler *CategoryHandler::getCopy()
...
-    response.setStatus(Poco::Net::HTTPServerResponse::HTTP_OK);
-    ostr<< jsonString.toStdString();
-    }
-    catch (...) {
-        response.setStatus(Poco::Net::HTTPServerResponse::HTTP_BAD_GATEWAY);
+        response.setStatus(HTTPServerResponse::HTTP_OK);
+        std::ostream& ostr = response.send();
+        ostr<< jsonString.toStdString();
+    } catch (...) {
+        response.setStatus(HTTPServerResponse::HTTP_BAD_GATEWAY);
```
Comment (AndriyFedorenko) [inline budget_manager_backend/sources/handlers/categoryhandler.cpp:45]: again here you should log something into log file

File: budget_manager_backend/sources/handlers/categoryhandler.cpp [hunk 6]
```diff
@@ -16,82 +16,93 @@ AbstractHandler *CategoryHandler::getCopy()
...
-    std::ostream& ostr = response.send();
-    response.setStatus(Poco::Net::HTTPServerResponse::HTTP_OK);
-    ostr<< jsonString.toStdString();
-    }
-    catch (...) {
-        response.setStatus(Poco::Net::HTTPServerResponse::HTTP_BAD_GATEWAY);
+        response.setStatus(HTTPServerResponse::HTTP_OK);
+        std::ostream& ostr = response.send();
+        ostr<< jsonString.toStdString();
+    } catch (...) {
```
Comment (AndriyFedorenko) [inline budget_manager_backend/sources/handlers/categoryhandler.cpp:44]: It is better to catch some specific exception and write log message from exception

File: budget_manager_frontend/headers/models/category.h [hunk 7]
```diff
@@ -9,23 +9,23 @@ class Category
...
     int getUserId();
     QString getName();
-    Type getType();
+    QString getType();
     QColor getColor();
 
     void setId(int id);
     void setName(QString name);
-    void setType(Type type);
+    void setType(QString type);
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/headers/models/category.h:28]: please add const and reference

File: budget_manager_frontend/headers/windows/addcategorydialog.h [hunk 8]
```diff
@@ -16,7 +16,7 @@ class AddCategoryDialog : public QDialog
     Q_OBJECT
 
 public:
-    AddCategoryDialog(int userId, Type type, QSharedPointer<QNetworkAccessManager> manager, QWidget *parent = nullptr);
+    AddCategoryDialog(int userId, QString type, QSharedPointer<QNetworkAccessManager> manager, QWidget *parent = nullptr);
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/headers/windows/addcategorydialog.h:19]: please add const and reference

File: budget_manager_frontend/sources/models/category.cpp [hunk 9]
```diff
@@ -1,15 +1,17 @@
...
     // Attention! Not valid data!
     this->id = -1;
     this->name = QString();
-    this->type = income;
+    this->type = INCOME;
     this->color = QColor();
 }
 
-Category::Category(int id, QString name, Type type, QColor color, int userId)
+Category::Category(int id, QString name, QString type, QColor color, int userId)
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/models/category.cpp:14]: please add const and reference

File: DAL/headers/models/user.h (hunk 1 above)
Comment (BogdanKalysh) [reply DAL/headers/models/user.h:12]: removed

File: DAL/sources/repositories/categoryrepository.cpp (hunk 3 above)
Comment (BogdanKalysh) [reply DAL/sources/repositories/categoryrepository.cpp:79]: Will use our logger soon

File: budget_manager_backend/sources/applicationservice.cpp (hunk 4 above)
Comment (BogdanKalysh) [reply budget_manager_backend/sources/applicationservice.cpp:33]: removed

File: budget_manager_backend/sources/handlers/categoryhandler.cpp (hunk 5 above)
Comment (BogdanKalysh) [reply budget_manager_backend/sources/handlers/categoryhandler.cpp:45]: Vill use logger

File: budget_manager_frontend/sources/windows/signUpWindow.cpp [hunk 10]
```diff
@@ -32,66 +34,57 @@ SignUpWindow::~SignUpWindow()
...
+        ui->emailLine->clear();
+        ui->nameLine->clear();
+        ui->passwordLine->clear();
+        ui->repeatPasswordLine->clear();
+
+        this->clearMask();
+        this->close();
+        emit loginWindow();
+    } else {
+        qDebug()<<postUserReply->error();
```
[the remark this answers is not in the document]
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/windows/signUpWindow.cpp:85]: Will use logger

File: DAL/sources/models/transaction.cpp (hunk 2 above)
Comment (BogdanKalysh) [reply DAL/sources/models/transaction.cpp:100]: used reference

File: budget_manager_frontend/sources/models/category.cpp (hunk 9 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/models/category.cpp:14]: used references

File: budget_manager_frontend/headers/models/category.h (hunk 7 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/headers/models/category.h:28]: used const reference everywhere

File: budget_manager_frontend/headers/windows/addcategorydialog.h (hunk 8 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/headers/windows/addcategorydialog.h:19]: used const reference everywhere

File: budget_manager_frontend/headers/windows/mainWindow.h [hunk 11]
```diff
@@ -26,7 +26,7 @@ class MainWindow : public QMainWindow
 
 protected:
     void configWindowItems();
-    qreal getCategoryTotalSum(QString categoryName, Type type);
+    qreal getCategoryTotalSum(QString categoryName, QString type);
```
[the remark this answers is not in the document]
Comment (BogdanKalysh) [reply budget_manager_frontend/headers/windows/mainWindow.h:29]: used const reference everywhere

**Pull request merged**